### ISSUE TITLE
Link to commits includes extra newline.

### ISSUE THREAD

**Issue by last-partizan** [mattermost/mattermost-plugin-gitlab#98]
Hello.

Right now push notify looks like this:
![Screenshot from 2019-09-18 15-23-35](https://user-images.githubusercontent.com/301015/65148201-6637f100-da28-11e9-8bba-97f01bd20786.png)

Is it becouse commit.Message contains unexpected trailing newline? is someone has similar issue, or is it just me?
@hanzei can you take a look at this?

https://github.com/mattermost/mattermost-plugin-gitlab/blob/4990fd6c59f7854bd62fc104b1659edd257583bf/server/webhook/push.go#L56

**Comment by hanzei** [comment]
@last-partizan Does your commit message return a line break?

**Comment by last-partizan** [comment]
@hanzei it has no linebreaks at all.

**Comment by hanzei** [comment]
The code looks fine to be and the test are also fine. @manland Do you have any idea why this is happening?

**Comment by manland** [comment]
@last-partizan have you experimented this in other commit? 

The [unit test](https://github.com/mattermost/mattermost-plugin-gitlab/blob/master/server/webhook/push_test.go#L26-L30) is good => no break line in commit == no break line in message.

**Comment by last-partizan** [comment]
@manland Yes, this happens in every commit.

Looks like in real life, push webhook contains extra newline in the end of the message (and can contain newlines in general).

Here is one my commit
```
  "commits": [
    {
      "id": "7d079c62ed79c703712ce4a3fddb3ec0a58e43ca",
      "message": "TODO cleanup\n",
      "timestamp": "2019-09-30T15:05:29Z",
...
```

And here is another:
```
  "commits": [
    {
      "id": "1f2f7b608ae935b1d40981a78f783fb7aeed5a5e",
      "message": "Update todo\n\n- Test newlines in message\n- Second line (no new line at the end)\n",
      "timestamp": "2019-09-30T15:17:12Z",
...
```

**Comment by octoquad** [comment]
I have this issue as well with Gitlab 12.3.5 and Mattermost 5.14.2. Commit messages have this trailing new line, but commit messages generated by Gitlab such as "Merge request #1 <Name> was merged by user" don't have the new line. 

Additionally, if formatting your commit messages like:

```
Description

* Fixed X
* Fixed Y

Resolves #123
```

The plugin formats it as 

```
Joe Coder has pushed 1 commit to namespace/repo
    * [Fixed X
    * Fixed Y
](https://url/to/commit)
```

From the mattermost log:
```
{"level":"info","ts":1570605416.749483,"caller":"mlog/sugar.go:19","msg":"new msg","plugin_id":"com.github.manland.mattermost-plugin-gitlab","message":"[Joe Coder](https://url/to/coder) has pushed 1 commit to [namespace/repo](https://url/to/rep)\n- [Fixed X\n* Fixed Y\n](https://url/to/commit)","from":"Joe Coder"}
```

The Gitlab hook JSON data also shows a newline, but the merge request as well as the commit message do not contain a new line. 

I'm going to assume, based off the data above, that Gitlab adds that in automatically.

**Comment by octoquad** [comment]
https://github.com/mattermost/mattermost-plugin-gitlab/blob/4990fd6c59f7854bd62fc104b1659edd257583bf/server/webhook/push.go#L56

Regarding the `\n-` in this line of code, this always assumes that the message is going to be a single line or paragraph. Perhaps it would be better to do:

```go
message += fmt.Sprintf("\n %s [%s](%s)", commit.Message, 'View Commit', commit.URL) 
```

Unfortunately, I'm not sure if the change above breaks any backwards compatibility with older versions of Gitlab.

**Comment by manland** [comment]
@octoquad nop, your right, it's a bug from this plugin. It will work with oldest version of gitlab because : 

* [commit](http://commit.com)

WIll become : 

* commit [view Commit](http://commit.com)

A PR is welcomed of course!

**Comment by octoquad** [comment]
@manland OK, I have zero experience with Go, but I'll give it a bash tomorrow :)

**Comment by last-partizan** [comment]
Maybe it's better to only use first line of commit? just like github/gitlab does for multiline commit messages.

It would be better than

```
- Someone pushed "Commit message

commit message next line" [View commit]
```

**Comment by octoquad** [comment]
> Maybe it's better to only use first line of commit? just like github/gitlab does for multiline commit messages.
> 
> It would be better than
> 
> ```
> - Someone pushed "Commit message
> 
> commit message next line" [View commit]
> ```

With the changes I pushed it will actually be
```
Bruce Pieterse has pushed 1 commit to namespace/repo
Refactor

Commit Description

* Bullet 1
* Bullet 2

Resolves #95
View Commit
```

For simpler messages it:

```
Bruce Pieterse has pushed 2 commits to namespace/repo
Changed X to Y
View Commit
Corrected typo in previous commit
View Commit
```
